### PR TITLE
Using a Spack build cache.

### PR DESCRIPTION
**Pull request by t-brown**
Updating the instructions so as to use a Spack build cache.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: content/02-Cluster/04-install-spack.md [hunk 1]
```diff
@@ -11,16 +11,24 @@ tags: ["tutorial", "pcluster-manager", "ParallelCluster", "Spack"]
...
 git clone -c feature.manyFiles=true https://github.com/spack/spack $SPACK_ROOT
-cd $SPACK_ROOT
-exit
 echo "export SPACK_ROOT=/shared/spack" >> $HOME/.bashrc
 echo "source \$SPACK_ROOT/share/spack/setup-env.sh" >> $HOME/.bashrc
 source $HOME/.bashrc
-sudo chown -R $USER:$USER $SPACK_ROOT
+```
+
+We are going to install the weather codes from a Spack binary build cache/mirror. In order to do this we need to install a few more python packages.
```
Comment (sean-smith) [inline content/02-Cluster/04-install-spack.md:21]: Let's add a note on why we're pinning boto3 and link to the Spack PR to fix it. Best to be transparent.

File: content/02-Cluster/04-install-spack.md [hunk 2]
```diff
@@ -11,16 +11,24 @@ tags: ["tutorial", "pcluster-manager", "ParallelCluster", "Spack"]
...
+
+```
+pip3 install botocore==1.23.46 boto3==1.20.46
+```
+
+Next we add the mirror and trust the GPG keys that have signed the packages.
+
+```
+spack mirror add aws-hpc-weather s3://aws-hpc-weather/spack/
+spack buildcache keys --install --trust --force
```
Comment (sean-smith) [inline content/02-Cluster/04-install-spack.md:31]: Is there a way we can post a key hash/public key somewhere where they can verify?

File: content/02-Cluster/04-install-spack.md [hunk 3]
```diff
@@ -11,16 +11,24 @@ tags: ["tutorial", "pcluster-manager", "ParallelCluster", "Spack"]
...
-mkdir -p $SPACK_ROOT
 git clone -c feature.manyFiles=true https://github.com/spack/spack $SPACK_ROOT
-cd $SPACK_ROOT
-exit
 echo "export SPACK_ROOT=/shared/spack" >> $HOME/.bashrc
 echo "source \$SPACK_ROOT/share/spack/setup-env.sh" >> $HOME/.bashrc
 source $HOME/.bashrc
-sudo chown -R $USER:$USER $SPACK_ROOT
+```
+
```
Comment (sean-smith) [inline content/02-Cluster/04-install-spack.md:20]: add a section header here

File: content/02-Cluster/04-install-spack.md (hunk 2 above)
Comment (t-brown) [reply content/02-Cluster/04-install-spack.md]: You mean like [here](https://keys.openpgp.org/search?q=aws-hpc-weather%40amazon.com). 😄

**Comment by sean-smith** [comment]
Change LGTM, I'm going to hold off on approving until we get the S3 Bucket sorted.

**Comment by sean-smith** [comment]
We need to add `s3://` permissions to the cluster configuration otherwise we see:

```bash
[ec2-user@ip-172-31-18-33 ~]$ spack mirror add aws-hpc-weather s3://aws-hpc-weather/spack/
[ec2-user@ip-172-31-18-33 ~]$ spack buildcache keys --install --trust --force
==> Error: An error occurred (AccessDenied) when calling the GetObject operation: Access Denied
```

**Review by sean-smith — APPROVED** [review]

**Pull request merged**